File: genium/src/test/resources/smoke/constants/output/android/com/example/smoke/ConstantsInterface.java
/*
 *

 */
package com.example.smoke;
import com.example.NativeBase;
public final class ConstantsInterface extends NativeBase {
    public static final boolean BOOL_CONSTANT = true;
    public static final int INT_CONSTANT = -11;
    public static final long UINT_CONSTANT = 4294967295L;
    public static final float FLOAT_CONSTANT = 2.71f;
    public static final double DOUBLE_CONSTANT = -3.14;
    public static final String STRING_CONSTANT = "Foo bar";
    public static final ConstantsInterface.StateEnum ENUM_CONSTANT = ConstantsInterface.StateEnum.ON;
    public enum StateEnum {
        OFF(0),
        ON(1);
        public final int value;
        StateEnum(final int value) {
            this.value = value;
        }
    }
    /**
     * For internal use only.
     * @exclude
     */
    protected ConstantsInterface(final long nativeHandle) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
}

File: gluecodium/src/test/resources/smoke/constants/output/android-kotlin/com/example/smoke/ConstantsInterface.kt
/*

 *
 */

@file:JvmName("ConstantsInterface")

package com.example.smoke

import com.example.NativeBase

class ConstantsInterface : NativeBase {

    enum class StateEnum(private val value: Int) {
        OFF(0),
        ON(1);
    }


    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmField final val BOOL_CONSTANT: Boolean = true
        @JvmField final val INT_CONSTANT: Int = -11
        @JvmField final val UINT_CONSTANT: Long = 4294967295L
        @JvmField final val FLOAT_CONSTANT: Float = 2.71f
        @JvmField final val DOUBLE_CONSTANT: Double = -3.14
        @JvmField final val STRING_CONSTANT: String = "Foo bar"
        @JvmField final val ENUM_CONSTANT: ConstantsInterface.StateEnum = ConstantsInterface.StateEnum.ON
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
